src/Codegen/Codegen.hs

```diff
@@ -664,8 +667,11 @@ genExpr (I.Match s as t) = do
 
   (cases, blks) <- bimap concat concat . unzip <$> mapM genArm as
   return (val, sStms ++ assignScrut ++ switch cases ++ blks ++ joinStm)
-genExpr I.Lambda{}      = fail "Cannot handle lambdas"
-genExpr (I.Prim p es t) = genPrim p es t
+genExpr I.Lambda{}        = fail "Cannot handle lambdas"
+genExpr (I.Prim p es t  ) = genPrim p es t
+genExpr (I.Exception _ t) = do
+  tmp <- genTmp t
+  return (tmp, [citems|$exp:(throw INTERNAL_ERROR);|]) -- unit instead of temp?
```

Comment:
Yeah, I think you can just return `unit` here instead of a temp variable

Reply:
Done